opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -911,10 +930,25 @@ namespace Opm {
             WellGroupHelpers::updateGuideRates(fieldGroup, schedule(), summaryState, this->phase_usage_, reportStepIdx, simulationTime,
                                         this->wellState(), this->groupState(), comm, &this->guideRate_, pot, local_deferredLogger);
         }
-        last_report_.converged = true;
-        last_report_.assemble_time_well += perfTimer.stop();
+
+        // Maybe do a recursive call to iterate network and well controls.
+        if (network_changed) {
```

Comment:
With the current way of handling the group control, even if the network wells are not under THP control, there still can be some well rate change with iterations. Should we still do more iterations even the `network_change` is `false` here?

Reply:
I think not. That should be separate logic.